src/System/TimedScopes/Correlation.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#region Using Directives
+
+using Microsoft.Omex.System.Context;
+using System;
+using Microsoft.Omex.System.Diagnostics;
+
+#endregion
+
+
+namespace Microsoft.Omex.System.TimedScopes
+{
+	/// <summary>
+	/// Correlation class, handle in-memory correlation data for logging purposes.
+	/// </summary>
+	/// <remarks>
+	/// Allows correlation data to be set and retrieved on a calling context. ULS logging (when hooked up)
+	/// only allows for the correlation id to be retrieved, this allows for additional data to be retrieved.
+	/// Correlation data by default only applies to the current thread and is lost when a thread hop
+	/// occurs (be it ASP.NET underlying thread hop or an explicit thread change). By calling
+	/// CorrelationStart with cloned CorrelationData, the correlation can be transferred to the new
+	/// thread.
+	/// </remarks>
+	public sealed class Correlation
+	{
+		/// <summary>
+		/// Constructor
+		/// </summary>
+		/// <param name="correlationHandler">correlation handler, used to store correlation data in underlying logging system</param>
+		/// <param name="callContextManager">call context manager, to handle context operations</param>
+		/// <param name="machineInformation">machine information</param>
+		public Correlation(ICorrelationStorage correlationHandler, ICallContextManager callContextManager, IMachineInformation machineInformation)
```

Comment:
>Correlation [](start = 9, length = 11)

Validate constructor parameters?

Reply:
Will this changes (Code.ExpectsArgument) be sufficient?